### PR TITLE
setProp -> setCssLength for 'left', 'top', 'right', 'bottom', 'width', 'height'

### PR DESCRIPTION
**Pull request by balazsbajorics**
**Problem:**
We have the somewhat overlapping setProperty and setCssLength commands. I wanted to make sure that for css length properties 'left', 'top', 'right', 'bottom', 'width', 'height' we use the setCssLength, and never setProperty.

**Fix:**
1. Type setProperty in a way to disallow PropertyPaths ending in 'left', 'top', 'right', 'bottom', 'width', 'height'.
2. Fix the problems by migrating them over to setCssLength
3. Profit

### CODE REVIEW

File: editor/src/utils/common-constants.ts [hunk 1]
```diff
@@ -1 +1 @@
-export const styleStringInArray = ['style']
+export const styleStringInArray = ['style'] as ['style']
```
Comment (balazsbajorics) [inline editor/src/utils/common-constants.ts:1]: this looks a bit wonky but it makes sure that the type is the array literal with a string literal inside it. this means then typescript can pick up this type where we use it to build longer paths!

File: editor/src/components/inspector/inspector-strategies/fill-container-basic-strategy.tsx [hunk 2]
```diff
@@ -29,21 +33,22 @@ export const fillContainerStrategyBasic = (
     }
 
     return elements.flatMap((path) => {
-      const parentInstance = MetadataUtils.findElementByElementPath(metadata, EP.parentPath(path))
-      if (!MetadataUtils.isFlexLayoutedContainer(parentInstance)) {
+      const instance = MetadataUtils.findElementByElementPath(metadata, path)
+      // TODO BEFORE MERGE!! this is an unrelated change I just fixed it
```
Comment (balazsbajorics) [inline editor/src/components/inspector/inspector-strategies/fill-container-basic-strategy.tsx:37]: @bkrmendy I think this one is for you. I noticed that this place was using EP.parentPath to check something and I _believe_ I replaced it with something equivalent but less error-prone. could you double check please?

Comment (bkrmendy) [reply editor/src/components/inspector/inspector-strategies/fill-container-basic-strategy.tsx:37]: the parent instance is needed to check whether it's a flex container, what you added is in fact a safer and better way of doing that so big +1 from me

File: editor/src/utils/common-constants.ts (hunk 1 above)
Comment (bkrmendy) [reply editor/src/utils/common-constants.ts:1]: is this the same as suing `as const`?

Comment (balazsbajorics) [reply editor/src/utils/common-constants.ts:1]: `as const` makes it a `ReadonlyArray<'style'>` and quite a few places complained that they take Array and not ReadonlyArray, and I didn't want to inflate the scope of this PR

**Review by bkrmendy — APPROVED** [review]

**Review by Rheeseyb — APPROVED** [review]

**Pull request merged**